imports/plugins/core/system-info/server/no-meteor/queries/systemInformation.js

```diff
@@ -0,0 +1,26 @@
+import packageJson from "/package.json";
+import ReactionError from "@reactioncommerce/reaction-error";
+
+/**
+ * @name queries.systemInformation
+ * @method
+ * @memberof SystemInformation/GraphQL
+ * @summary get systemInformation
+ * @param {Object} context - an object containing the per-request state
+ * @param {String} shopId The ID of the shop to load settings for
+ * @return {Promise<Object>} System Information
+ **/
+export default async function systemInformation(context, shopId) {
+  const { userHasPermission, collections: { Packages }, app: { db } } = context;
+  // sensitive information should be accesible to admins only
+  if (!userHasPermission(["admin"], shopId)) throw new ReactionError("access-denied", "User does not have permission");
+
+  const mongoAdmin = await db.admin();
+  const mongoInfo = await mongoAdmin.serverStatus();
+  const plugins = await Packages.find({ shopId: { shopId }, version: { $ne: null } }, { projection: { name: 1, version: 1 } }).toArray();
```

Comment:
Since initially reviewing this, we've been talking more about trying to stop using the `Packages` collection. The problem is that the collection can get out of sync with what plugins are truly in the codebase. So it would be better to access the actual list of plugins that were registered when the app started up.

On context, `context.app` is the `ReactionNodeApp` instance. On that you can find the plugin config for all plugins on `app.registeredPlugins` object. So I think it would be something like this:

```js
const plugins = Object.values(context.app.registeredPlugins);
```

The rest should be the same.

Reply:
Note that unlike the `Packages` collection, `registeredPlugins` is NOT different per shopId. But I think you should keep the shopId param in GraphQL for the permission check and potential future use.